tests/mypy/outputs/plugin-success.txt

```diff
@@ -0,0 +1,3 @@
+121: error: Unexpected keyword argument "name" for "AddProject"  [call-arg]
```

Comment:
Weird -- is the dataclass_class_maker_callback not properly adding fields?

Reply:
Sorry, I might make a mistake with the filename.
The result has not used the plugin.
Should I change the name to `fail4.txt`
```
    ('mypy-default.ini', 'fail3.py', 'fail3.txt'),
    ('mypy-default.ini', 'plugin_success.py', 'plugin-success.txt'),
```

Reply:
Ohh I see.

I think your naming is fine -- I originally named it based on whether the file would execute successfully, not whether the plugin would have zero errors.

At any rate, I wonder if there might be a way to get mypy to treat pydantic dataclasses like regular dataclasses if the plugin is NOT installed, and only override the behavior if the plugin is active. That way it won't break mypy for anyone currently using pydantic dataclasses that don't want to deal with plugins.

But I suppose if it isn't possible it would just push more people to use the plugin, which is maybe a good thing anyway.

Reply:
I agree.
I have deleted the logic. But, we may keep it to get same result by mypy. I will think about it tomorrow.
https://github.com/samuelcolvin/pydantic/blob/0c18619769e3b4a975dea01f6e7f189a15639863/pydantic/dataclasses.py#L108-L112

Reply:
Also, the mypy devs are telling me on gitter it's probably not possible to have it resolve to the `dataclasses.dataclass` decorator with the plugin off, but with our custom behavior with the plugin on.

I'll let you know if I hear anything different, but I guess if we want to add introspection of the pydantic-specific `dataclass` decorator, we'll probably just need to force people to use the mypy plugin. It's probably for the best to get more plugin usage anyway.

Reply:
sounds good to me.